### PR TITLE
Add tf2paddle to convert models

### LINKED ISSUE #482: How to use a model from TF
How to use a model from TF

### PR DESCRIPTION
**Pull request by guoshengCS**
Fix #482

### CODE REVIEW

File: image_classification/tf2paddle/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+## 使用说明
+
+`tf2paddle.py`提供了将TensorFlow训练的模型转换为PaddlePaddle可使用的模型的接口`TFModelConverter`，其封装了图像领域常用的Convolution、BatchNorm等layer的转换函数，可以完成VGG、ResNet等常用模型的转换。模型转换的基本过程是：基于TensorFlow的Python API获取variable，将各variable对应到PaddlePaddle中layer的参数，进行适配后序列化保存输出可以直接为PaddlePaddle的Python API加载使用的模型文件。
+
+为使TensorFlow模型中的variable能够正确对应到PaddlePaddle模型中layer的参数，正确完成转换，模型转换具有如下约束：
+
+- 支持TensorFlow中conv2d，batchnorm，fc这三种带有trainable variable的Operator中参数的转换。- TensorFlow配置中同一Operator内的variable属于相同的scope，以此将variable划分到不同的layer。
```
Comment (jacquesqiao) [inline image_classification/tf2paddle/README.md]: rm ^M

Comment (lcy-seso) [reply image_classification/tf2paddle/README.md]: Done.

File: image_classification/tf2paddle/README.md [hunk 2]
```diff
@@ -1,27 +1,40 @@
...
-    build_graph()
-    sess = tf.Session(config=tf.ConfigProto(allow_soft_placement=True))
-    sess.run(tf.tables_initializer())
-    saver = tf.train.Saver()
-    saver.restore(sess, 'model/model.ckpt')
-    return sess
-```
+1. 目前仅支持将TensorFlow中 `conv2d`，`batchnorm`，`fc`这三种带有可学习`Variable`的Operator训练出的参数向PaddlePaddle模型参数转换。
+1. TensorFlow网络配置中同一Operator内的`Variable`属于相同的scope，以此为依据将`Variable`划分到不同的`paddle.layer`。
+1. `conv2d`、`batchnorm`、`fc`的scope需分别包含`conv`、`bn`、`fc`，以此获取对应`paddle.layer`的类型。也可以通过为`TFModelConverter`传入`layer_type_map`的`dict`，将scope映射到对应的`paddle.layer`的type来规避此项约束。
```
Comment (lcy-seso) [inline image_classification/tf2paddle/README.md]: 这里的 `conv2d`、`batchnorm`、`fc`  能否换成精确的 TensorFlow 中的调用接口，比如 ：`conv2d` --> `tf.nn.conv2d`，方便使用者在看到文档时候，能够精确定位到自己配置中的 operator。

File: image_classification/tf2paddle/README.md [hunk 3]
```diff
@@ -1,27 +1,40 @@
...
+1. `conv2d`、`fc`中`Variable`的顺序为：先可学习`Weight`后`Bias`；`batchnorm`中`Variable`的顺序为：`scale`、`shift`、`mean`、`var`，请注意参数存储的顺序将`Variable`对应到`paddle.layer.batch_norm`相应位置的参数。
```
Comment (lcy-seso) [inline image_classification/tf2paddle/README.md]: 同上，请把 `conv2d`、`batchnorm`、`fc`  换成精确的 TensorFlow 中的调用接口。

File: image_classification/tf2paddle/README.md [hunk 4]
```diff
@@ -1,27 +1,40 @@
...
+1. TensorFlow网络拓扑顺序需和PaddlePaddle网络拓扑顺序一致，尤其注意具有分支时左右分支的顺序。这是针对模型转换和PaddlePaddle网络配置均使用PaddlePaddle默认参数命名的情况，此时将根据拓扑顺序进行参数命名。
```
Comment (lcy-seso) [inline image_classification/tf2paddle/README.md]: 这句话我不是非常理解 “尤其注意具有分支时左右分支的顺序”，能否进一步解释一下。这里值的“左右分支”是什么呢？

**Review by lcy-seso — APPROVED** [review]
LGTM
